src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3731,6 +3747,32 @@ theorem getLsbD_twoPow (i j : Nat) : (twoPow w i).getLsbD j = ((i < w) && (i = j
           simp at hi
         simp_all
 
+@[simp]
+theorem msb_twoPow {i w: Nat} :
+    (twoPow w i).msb = (decide (i < w) && decide (i = w - 1)) := by
+  simp only [BitVec.msb, getMsbD_eq_getLsbD, Nat.sub_zero, getLsbD_twoPow,
+    Bool.and_iff_right_iff_imp, Bool.and_eq_true, decide_eq_true_eq, and_imp]
+  intros
+  omega
+
+theorem toInt_twoPow {w i : Nat} :
+    (BitVec.twoPow w i).toInt = if w ≤ i then 0 else (2^i : Nat) * if i + 1 = w then (-1 : Int) else 1 := by
```

Comment:
```suggestion
theorem toInt_twoPow {w i : Nat} :
    (BitVec.twoPow w i).toInt =
    (if w ≤ i then 0 else (2^i : Nat)) * 
    (if i + 1 = w then (-1 : Int) else 1) := by
```

I know I was the one who suggested this, but I'm not certain what the perfect choice is, between the statement I proposed above, and the perhaps easier to think about:

```lean
theorem toInt_twoPow {w i : Nat} :
    (BitVec.twoPow w i).toInt =
    (if w ≤ i then 0
    else if i + 1 = w then 2^i else - 2^i := by 
```

Reply:
I can offer:

```lean
theorem toInt_twoPow {w i : Nat} :
    (BitVec.twoPow w i).toInt = if w ≤ i then 0 else if i + 1 = w then (-(2^i : Nat) : Int) else 2^i := by
  simp only [BitVec.toInt_eq_msb_cond, toNat_twoPow_eq_if]
  rcases w with _|w
  · simp
  · by_cases h : i = w
    · simp [h, show ¬ w + 1 ≤ w by omega]
      omega
    · by_cases h' : w + 1 ≤ i
      · simp [h', show ¬ i < w + 1 by omega]
      · simp [h, h', show i < w + 1 by omega, Int.natCast_pow]
```

Reply:
I pushed this one.